compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1146,12 +1146,16 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
       else {
         val flip = new TypeMap {
           def apply(t: Type) = t match {
-            case t: TypeAlias if variance > 0 && t.variance < 0 => t.derivedTypeAlias(t.alias, 1)
             case t: TypeBounds => t
+            case t @ AppliedType(tycon, args) =>
```

Comment:
Complication: The `flip` hack got now slightly trickier because we can't play with TypeAlias variances anymore.